.github/workflows/desktop.yml

```diff
@@ -0,0 +1,160 @@
+name: Desktop Builds
+
+on: 
+  push:
+  pull_request:
+
+env:
+  CCACHE_DIR: ${{ github.workspace }}/ccache_dir
+
+jobs:
+  build:
+    name: ${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.architecture }}-${{ matrix.python_version }}-${{ matrix.generator }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
+        build_type: ["Release"]
+        architecture: ["x64",]
+        python_version: [2.7]
+        generator: ["Visual Studio 16 2019", "Unix Makefiles", "Ninja"]
+        exclude:
+        - os: ubuntu-latest
+          generator: "Visual Studio 16 2019"
+        - os: macos-latest
+          generator: "Visual Studio 16 2019"
+        - os: windows-latest
+          generator: "Unix Makefiles"
+        - os: windows-latest
+          generator: "Ninja"
+
+        include:
+        - os: windows-latest
+          architecture: "x64"
+          vcpkg_triplet: "x64-windows-static"
+          msbuild_platform: "x64"
+        - os: ubuntu-latest
+          architecture: "x64"
+          vcpkg_triplet: "x64-linux"
+        - os: macos-latest
+          architecture: "x64"
+          vcpkg_triplet: "x64-osx"
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: true
+
+      - name: Set env variables for subsequent steps (all)
+        run: |
+          echo "::set-env name=VCPKG_RESPONSE_FILE::external/vcpkg_${{ matrix.vcpkg_triplet }}_response_file.txt"
+          echo "::set-env name=PIP_REQUIREMENTS_FILE::external/pip_requirements.txt"
+          echo "::set-env name=MATRIX_UNIQUE_NAME::${{ matrix.os }}-${{ matrix.build_type }}-${{ matrix.architecture }}-${{ matrix.python_version }}-${{ matrix.generator }}"
+          echo "test if ccache cache works"
+
+      - name: Install Ninja (windows)
+        if: startsWith(matrix.generator, 'Ninja') && startsWith(matrix.os, 'windows')
+        run: |
+          choco install ninja
+
+      - name: Install Ninja (mac)
+        if: startsWith(matrix.generator, 'Ninja') && startsWith(matrix.os, 'macos')
+        run: |
+          brew install ninja
+
+      - name: Install Ninja (linux)
+        if: startsWith(matrix.generator, 'Ninja') && startsWith(matrix.os, 'ubuntu')
+        run: |
+          sudo apt install ninja-build
+
+      - name: Install system level applications (mac)
+        if: startsWith(matrix.os, 'macos')
+        id: macos-brew-install
+        run: |
+          brew install ccache
+
+      - name: Install system level applications (linux)
+        if: startsWith(matrix.os, 'ubuntu')
+        id: ubuntu-brew-install
```

Comment:
should this be "apt" not "brew"?

Reply:
Yup! Good catch :)! I copy/pasted the mac block and forgot to fix the "id"